airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py

```
@@ -157,60 +167,37 @@ def close_partition(self, partition: Partition) -> None:
-                # If any earlier partition is still not finished, we must stop
-                if p_key not in self._finished_partitions or sem._value != 0:
-                    all_left_finished = False
-                    break
-                # Once we've reached earliest_key in the semaphore order, we can stop checking
-                if p_key == earliest_key:
-                    break
-
-            # If the partitions up to earliest_key are not all finished, break the while-loop
-            if not all_left_finished:
-                break
+            earliest_key, (candidate_state, candidate_seq) = next(
```

Pylint (3.3.7)

[warning] Unused variable 'earliest_key'

(W0612)